include/eve/detail/function/simd/arm/neon/swizzle.hpp

```diff
@@ -0,0 +1,89 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright 2018-2021 Joel FALCOU
+  Copyright 2018-2021 Jean-Thierry LAPRESTE
+
+  Licensed under the MIT License <http://opensource.org/licenses/MIT>.
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+#pragma once
+
+#include <eve/detail/abi.hpp>
+#include <eve/detail/function/simd/common/swizzle_helpers.hpp>
+#include <eve/forward.hpp>
+#include <eve/pattern.hpp>
+
+namespace eve::detail
+{
+  template<typename T, typename N, arm_abi ABI, shuffle_pattern Pattern>
+  EVE_FORCEINLINE auto basic_swizzle_( EVE_SUPPORTS(neon128_), wide<T,N,ABI> const& v, Pattern const&)
+  {
+    constexpr auto sz = Pattern::size(N::value);
+    using that_t      = as_wide_t<wide<T,N,ABI>,fixed<sz>>;
+
+    constexpr Pattern q = {};
+
+    // We're swizzling so much we aggregate the output
+    if constexpr( has_aggregated_abi_v<that_t> )
+    {
+      return aggregate_swizzle(v,q);
+    }
+    else
+    {
+      if constexpr( std::same_as<ABI, arm_64_> )
+      {
+        using that_abi_t  = typename that_t::abi_type;
+        using bytes_t     = typename that_t::template rebind<std::uint8_t,fixed<8>>;
+        auto b0 = bit_cast(v,as_<bytes_t>());
+
+        if constexpr( std::same_as<that_abi_t, arm_64_> )
+        {
+          bytes_t that = vtbl1_u8(b0,as_bytes<wide<T,N,ABI>>(q,as_<bytes_t>()));
+          return bit_cast(that,as_<that_t>());
```

Comment:
How does swizzle change abi? I am confused:
can you swizzle to a different size/cardinality/type or not?
If you can, broadcast pattern and what not seem broken.
I don't see at all how swizzling to a different size works.